src/sentry/dynamic_sampling/rules/utils.py

```diff
@@ -192,15 +193,18 @@ def get_rule_type(rule: Rule) -> RuleType | None:
 def get_rule_hash(rule: PolymorphicRule) -> int:
     # We want to be explicit in what we use for computing the hash. In addition, we need to remove certain fields like
     # the sampleRate.
-    return json.dumps(
-        _deep_sorted(
+    return (
+        orjson.dumps(
             {
                 "id": rule["id"],
                 "type": rule["type"],
                 "condition": rule["condition"],
-            }
+            },
+            option=orjson.OPT_SORT_KEYS,
         )
-    ).__hash__()
+        .decode()
+        .__hash__()
```

Comment:
this should be replaced with a call to `hash(...)` -- calling the double-under directly is usually not recommended